### PR TITLE
Small fixes for PR 1522

### PR DESCRIPTION
**Pull request by alan-kut**

### CODE REVIEW

File: pkg/neg/syncers/endpoints_calculator_test.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ import (
 func TestLocalGetEndpointSet(t *testing.T) {
 	t.Parallel()
 	mode := negtypes.L4LocalMode
+	// The "enableEndpointSlices=false" case is enough since this test does not depend on endpoints at all.
```
Comment (prameshj) [inline pkg/neg/syncers/endpoints_calculator_test.go:38]: This does depend on endpoints to pick up the Subset.

File: pkg/utils/endpointslices/index_test.go [hunk 2]
```diff
@@ -60,12 +61,15 @@ func TestEndpointSlicesByServiceFunc(t *testing.T) {
 	for _, tc := range testCases {
 		t.Run(tc.desc, func(t *testing.T) {
 			keys, e := EndpointSlicesByServiceFunc(tc.obj)
-			if len(keys) != len(tc.expectedKeys) || (len(tc.expectedKeys) == 1 && keys[0] != tc.expectedKeys[0]) {
-				t.Errorf("Incorrect keys, got: %s, expected %s", keys, tc.expectedKeys)
+			if tc.expectError && e == nil {
```
Comment (prameshj) [inline pkg/utils/endpointslices/index_test.go:64]: nit: optional, but you could do "gotError := e != nil" and check if tc.expectError and gotError are equal.

**Review by prameshj — COMMENTED** [review]
/ok-to-test

looks good, except for one comment about Local traffic policy subsets test.

**Comment by freehan** [comment]
So this PR from OSS community member was trying to fix the bug. https://github.com/kubernetes/ingress-gce/pull/1462

This one adds a field ServicePortName field in ServicePort struct, which is missing for the translation.

File: pkg/neg/syncers/endpoints_calculator_test.go (hunk 1 above)
Comment (alan-kut) [reply pkg/neg/syncers/endpoints_calculator_test.go:38]: Done

File: pkg/utils/endpointslices/index_test.go (hunk 2 above)
Comment (alan-kut) [reply pkg/utils/endpointslices/index_test.go:64]: I don't see any clear way how to do this. Also the error message may change. I'd leave it as it is.

**Comment by alan-kut** [comment]
I fixed the comment about Local traffic policy subsets test.

File: pkg/neg/syncers/endpoints_calculator_test.go [hunk 3]
```diff
@@ -35,6 +35,7 @@ import (
 func TestLocalGetEndpointSet(t *testing.T) {
 	t.Parallel()
 	mode := negtypes.L4LocalMode
+	// The "enableEndpointSlices=false" case is enough since this test does not depend on endpoints to pick up the Subsets.
```
Comment (prameshj) [inline pkg/neg/syncers/endpoints_calculator_test.go:38]: This test does depend on endpoints to pick up the subsets. Subsets will only include nodes that have endpoints running and are valid LB candidates. We should test enableEndpointSlices=true here.

Other than this comment, the changes LGTM.

File: pkg/neg/syncers/utils.go [hunk 4]
```diff
@@ -261,10 +260,6 @@ func toZoneNetworkEndpointMap(eds []negtypes.EndpointsData, zoneGetter negtypes.
 				klog.V(2).Infof("Endpoint %q in Endpoints %s/%s does not have an associated node. Skipping", endpointAddress.Addresses, ed.Meta.Namespace, ed.Meta.Name)
 				continue
 			}
-			if endpointAddress.TargetRef == nil {
```
Comment (prameshj) [inline pkg/neg/syncers/utils.go:264]: Actually, I think we need this.. the previous check for TargerRef is inside an `if subsetLabels != ""`

File: pkg/neg/syncers/endpoints_calculator_test.go (hunk 3 above)
Comment (alan-kut) [reply pkg/neg/syncers/endpoints_calculator_test.go:38]: Done

File: pkg/neg/syncers/utils.go (hunk 4 above)
Comment (alan-kut) [reply pkg/neg/syncers/utils.go:264]: You are right, fixed.

Comment (alan-kut) [reply pkg/neg/syncers/utils.go:264]: Shouldn't it contain check: endpointAddress.TargetRef.Kind != "Pod" as well?

**Pull request merged**